otherlibs/unix/getlogin.c

```diff
@@ -13,17 +13,36 @@
 /*                                                                        */
 /**************************************************************************/
 
-#include <caml/mlvalues.h>
 #include <caml/alloc.h>
 #include "caml/unixsupport.h"
+#include <limits.h>
+#include <unistd.h>
 #include <errno.h>
 
-extern char * getlogin(void);
-
 CAMLprim value caml_unix_getlogin(value unit)
 {
-  char * name;
+  char *name;
+#ifdef HAVE_GETLOGIN_R
+  size_t bufsize;
+#ifdef LOGIN_NAME_MAX
+  bufsize = LOGIN_NAME_MAX;
+#else
+  long initlen = sysconf(_SC_LOGIN_NAME_MAX);
+  bufsize = initlen <= 0 ? _POSIX_LOGIN_NAME_MAX : (size_t) initlen;
+#endif
+  name = caml_stat_alloc(initlen);
```

Comment:
Are all combinations of `#ifdef` exercised in tests? It seems to me that if `HAVE_GETLOGIN_R` and `LOGIN_NAME_MAX` is defined, then `initlen` will not be defined on this line